nixos/modules/services/audio/vosk.nix

```diff
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.vosk;
+in
+{
+  options.services.vosk = with lib; {
+    enable = mkEnableOption "the Vosk WebSocket speech recognition server";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.vosk-server;
+      defaultText = literalExpression "pkgs.vosk-server";
+      description = "The package to use for the Vosk server.";
+    };
+    interface = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
```

Comment:
This is more of a secure default even if firewall is disabled. Otherwise, if we really want this: `if config.networking.firewall.enable then "0.0.0.0" else "127.0.0.1"` should be the good option.

Reply:
We shouldn't do such a confusing conditional.

Reply:
Therefore, we should have `127.0.0.1` by default IMHO.